test/gurka/test_ramps_tc.cc

```diff
@@ -279,3 +279,148 @@ TEST(RampsTCs, test_tc_infer) {
       result, maneuver_index, "Turn left onto KJ.", "Turn left onto KJ.",
       "Turn left onto KJ. Then You will arrive at your destination.", "Continue for 50 meters.");
 }
+
+namespace {
+void check_edge_classification(baldr::GraphReader& graph_reader,
+                               const gurka::nodelayout& nodes,
+                               const std::string& b,
+                               const std::string& e,
+                               baldr::RoadClass rc) {
+  const auto edge = std::get<1>(gurka::findEdgeByNodes(graph_reader, nodes, b, e));
+  EXPECT_EQ(edge->classification(), rc);
+}
+} // namespace
+
+TEST(LinkReclassification, test_use_refs) {
+  constexpr double gridsize_metres = 10;
+
+  const std::string ascii_map = R"(
+  A------------B-------------C-----------D
+                \     G     /
+                 \    |    K----------L
+     I------------J   |   /
+                   \  |  /
+                    \ | /
+                     \|/
+                      F
+                      |
+                      |
+                      E
+  )";
+
+  const gurka::ways ways = {
+      // motorway road M1
+      {"ABCD", {{"highway", "motorway"}, {"ref", "M1"}}},
+      // trunk road M2
+      {"EFG", {{"highway", "trunk"}, {"ref", "M2"}}},
+      // link from M1 to M2
+      {"BJF", {{"highway", "motorway_link"}, {"destination:ref", "M2"}, {"oneway", "yes"}}},
+      // some tertiary road
+      {"IJ", {{"highway", "tertiary"}}},
+      // link from M2 to M1
+      {"FKC", {{"highway", "trunk_link"}, {"ref", "M2"}, {"oneway", "yes"}}},
+      // some tertiary road
+      {"KL", {{"highway", "tertiary"}}},
+  };
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize_metres);
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_ramps_use_refs",
+                               {{"mjolnir.reclassify_links", "true"}});
+  baldr::GraphReader graph_reader(map.config.get_child("mjolnir"));
+
+  // check that ramp classification from M1 to M2 wasn't confused by the tertiary road
+  check_edge_classification(graph_reader, layout, "B", "J", baldr::RoadClass::kTrunk);
+  check_edge_classification(graph_reader, layout, "J", "F", baldr::RoadClass::kTrunk);
+
+  // check that ramp classification from M2 to M1 wasn't confused by the tertiary road
+  check_edge_classification(graph_reader, layout, "F", "K", baldr::RoadClass::kTrunk);
+  check_edge_classification(graph_reader, layout, "K", "C", baldr::RoadClass::kTrunk);
+}
+
+TEST(LinkReclassification, test_hierarchical_reclass) {
```

Comment:
check that parent links have higher (or equal) classification than their children